### ISSUE TITLE
Broken Pipe after check and brevent

### ISSUE THREAD

**Issue by aviraxp** [brevent/Brevent#424]
<details>
<summary>Click to show logs</summary>

```
08-31 01:53:09.713 I/BreventServer(5790): check and brevent
08-31 01:53:09.827 W/FastPrintWriter(994): Write failure
08-31 01:53:09.827 W/FastPrintWriter(994): java.io.IOException: write failed: EPIPE (Broken pipe)
08-31 01:53:09.827 W/FastPrintWriter(994): 	at libcore.io.IoBridge.write(IoBridge.java:531)
08-31 01:53:09.827 W/FastPrintWriter(994): 	at java.io.FileOutputStream.write(FileOutputStream.java:381)
08-31 01:53:09.827 W/FastPrintWriter(994): 	at com.android.internal.util.FastPrintWriter.flushBytesLocked(FastPrintWriter.java:352)
08-31 01:53:09.827 W/FastPrintWriter(994): 	at com.android.internal.util.FastPrintWriter.flushLocked(FastPrintWriter.java:375)
08-31 01:53:09.827 W/FastPrintWriter(994): 	at com.android.internal.util.FastPrintWriter.flush(FastPrintWriter.java:410)
08-31 01:53:09.827 W/FastPrintWriter(994): 	at android.os.Binder.dump(Binder.java:563)
08-31 01:53:09.827 W/FastPrintWriter(994): 	at android.os.Binder.onTransact(Binder.java:511)
08-31 01:53:09.827 W/FastPrintWriter(994): 	at android.app.IActivityManager$Stub.onTransact(IActivityManager.java:3592)
08-31 01:53:09.827 W/FastPrintWriter(994): 	at com.android.server.am.ActivityManagerService.onTransact(ActivityManagerService.java:3288)
08-31 01:53:09.827 W/FastPrintWriter(994): 	at android.os.Binder.execTransact(Binder.java:731)
08-31 01:53:09.827 W/FastPrintWriter(994): Caused by: android.system.ErrnoException: write failed: EPIPE (Broken pipe)
08-31 01:53:09.827 W/FastPrintWriter(994): 	at libcore.io.Linux.writeBytes(Native Method)
08-31 01:53:09.827 W/FastPrintWriter(994): 	at libcore.io.Linux.write(Linux.java:288)
08-31 01:53:09.827 W/FastPrintWriter(994): 	at libcore.io.BlockGuardOs.write(BlockGuardOs.java:348)
08-31 01:53:09.827 W/FastPrintWriter(994): 	at libcore.io.IoBridge.write(IoBridge.java:526)
08-31 01:53:09.827 W/FastPrintWriter(994): 	... 9 more
08-31 01:53:09.829 I/ActivityManager(994): Force stopping com.dmzj.manhua appid=10090 user=0: from pid 5790
08-31 01:53:09.830 I/ActivityManager(994): Killing 9827:com.dmzj.manhua/u0a90 (adj 900): stop com.dmzj.manhua
```
</details>

Latest version 3.4.1.1 with self built Android Pie on Oneplus2. Brevent actually works but this warning seems strange. I suppose I misses some binder commits in kernel.

**Comment by liudongmiao** [comment]
Since 3.4.X, Brevent Server implements pipe for dumpsys.

In previous version, Brevent Server waits for the completion of dumpsys, and put the results into a byte array, then parse them. This will cause two problem:

- it's unnecessary to cache the results of dumpsys into memory. and if the results is large, it will be oom like #416 

- brevent server don't need all the data. for example, if dumpsys returns large data, brevent only need some of them, then brevent would stop receive data, this will be fast, but may cause exception.

You can refer to /sdcard/Android/data/me.piebridge.brevent/, it will show dumpsys in the logs, you can check the service and arguments, then dumpsys manually to check the time cost.

**Comment by liudongmiao** [comment]
fix in 3.5.2
